lib/shopping_cart.rb

```diff
@@ -0,0 +1,49 @@
+require './lib/product'
+
+class ShoppingCart
+  attr_reader :name, :capacity, :products, :total_number_of_products, :products_by_category
+
+  def initialize(name, capacity)
+    @name = name
+    @capacity = capacity.to_i
+    @products = []
+    @details = {name: @name, capacity: @capacity}
+    @total_number_of_products = 0
+    @products_by_category = {}
+  end
+
+  def add_product(product)
+    @products << product
+    @total_number_of_products += product.quantity
+  end
+
+  def is_full?
+    if @total_number_of_products >= @capacity
+      true
+    else
+      false
+    end
```

Comment:
no need for this if/else statement. line 21 without the `if` will automatically return true or false depending on how the comparison evaluates.

Reply:
Also, I think you'll need to use a > instead of a >= according to the interaction pattern.